### PR TITLE
feat(demultiplexing) Add check for demultiplexing on on-deck dragen

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
NovaSeq x flowcell are synced into the `flow_cells` directory, and given that they will not have a `demuxstarted.txt` file there will be a time window where they get picked up by the demultiplexing automation.

Since they are getting demultiplexed on the NovaSeq x  there is no need to try and demultiplex them on the cluster.

This PR adds a second check in `is_demultiplexing_possible` which checks if the demultiplexing has started on-deck and if so, skips the flow cell.

### Added

- New check `has_demultiplexing_started_on_deck`
- Tests for `has_demultiplexing_started_on_deck`
- Tests for `has_demultiplexing_started_locally`

### Changed
- New entry point to the CLI after removal of weird import

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b 2369-fix-has-demux-started
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/__init__.py [hunk 1]
```diff
@@ -1 +0,0 @@
-from .base import base
```
Comment (Vince-janv) [inline cg/cli/__init__.py:1]: If anyone knows what this does please enlighten me. It was the root of a nasty circular import, and all tests pass when removing it.

File: cg/models/demultiplex/flow_cell.py [hunk 2]
```diff
@@ -210,10 +211,19 @@ def validate_flow_cell_dir_name(self) -> None:
             LOG.warning(message)
             raise FlowCellError(message)
 
-    def is_demultiplexing_started(self) -> bool:
-        """Check if demultiplexing started path exists."""
+    def has_demultiplexing_started_locally(self) -> bool:
+        """Check if demultiplexing has started path exists on the cluster."""
         return self.demultiplexing_started_path.exists()
 
+    def has_demultiplexing_started_on_deck(self) -> bool:
```
Comment (ChrOertlin) [inline cg/models/demultiplex/flow_cell.py:218]:
```suggestion
    def has_demultiplexing_started_on_sequencer(self) -> bool:
```
more intuitive on location like this

File: cg/models/demultiplex/flow_cell.py [hunk 3]
```diff
@@ -210,10 +211,19 @@ def validate_flow_cell_dir_name(self) -> None:
...
+        """Check if demultiplexing has started on the NovaSeq x machine."""
```
Comment (ChrOertlin) [inline cg/models/demultiplex/flow_cell.py:219]:
```suggestion
        """Check if demultiplexing has started on the NovaSeqX machine."""
```

File: cg/models/demultiplex/flow_cell.py [hunk 4]
```diff
@@ -210,10 +211,19 @@ def validate_flow_cell_dir_name(self) -> None:
...
+        latest_analysis: Path = get_latest_analysis_path(self.path)
+        if not latest_analysis:
+            return False
+        return Path(
+            latest_analysis, DemultiplexingDirsAndFiles.DATA, DemultiplexingDirsAndFiles.BCL_CONVERT
+        ).exists()
```
Comment (ChrOertlin) [inline cg/models/demultiplex/flow_cell.py:225]:
```suggestion
        latest_analysis: Path = get_latest_analysis_path(self.path)
        if not latest_analysis:
            return False
        return Path(
            latest_analysis, DemultiplexingDirsAndFiles.DATA, DemultiplexingDirsAndFiles.BCL_CONVERT
        ).exists()
```

Is this not checking locally whether the demultiplexing has finished on the sequencer?
I believe there is a "SecondaryAnalysisComplete.txt" file when these analysis are finished written. Maybe that is a better indicator? I think the enum is: `ANALYSIS_COMPLETED`

**Review by ChrOertlin — APPROVED** [review]
Looks good, just a question on what we check for when the demultiplexing happened on the sequencer.

File: cg/models/demultiplex/flow_cell.py (hunk 4 above)
Comment (Vince-janv) [reply cg/models/demultiplex/flow_cell.py:225]: The problem is that there might be a time window where sequencing is complete but demultiplexing (on deck) is not, so we want to know whether it is running at all, thus looking for the `BCLConvert` folder.

**Review by henrikstranneheim — APPROVED** [review]
Well done 👍

**Comment by Vince-janv** [comment]
### Deployed in stage and production

**Pull request merged**